python/pyarrow/tests/test_scalars.py

```diff
@@ -577,6 +607,13 @@ def test_binary(value, ty, scalar_typ):
     assert memview.strides == (1,)
 
 
+def test_binary_null():
+    s = pa.scalar(None, type=pa.binary())
+    assert s.as_py() is None
+    with pytest.raises(TypeError):
+        memoryview(s)
```

Comment:
This part can be moved in the `test_binary` with type being parametrised with `ty` parameter.

What about an empty string scalar `pa.scalar("", type=pa.binary())`? Can we add this to the test also?
I think this should hold:

```python 
assert buffer.len == 0
assert memview == ""
```

Reply:
Added in!